lib/ConsumerImpl.cc

```diff
@@ -1508,18 +1508,33 @@ void ConsumerImpl::hasMessageAvailableAsync(HasMessageAvailableCallback callback
 
     if (messageId == MessageId::latest()) {
         lock.unlock();
-        getLastMessageIdAsync([callback](Result result, const GetLastMessageIdResponse& response) {
+        getLastMessageIdAsync([this, callback](Result result, const GetLastMessageIdResponse& response) {
```

Comment:
You should capture the `shared_ptr` or `weak_ptr` in case the `ConsumerImpl` instance (that `this` points to) has been destroyed when the callback of `getLastMessageIdAsync` is called.